Kernel/Arch/aarch64/RPi/AUX.h

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) 2024, Sönke Holz <sholz8530@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <Kernel/Memory/TypedMapping.h>
+
+namespace Kernel::RPi {
+
+struct AUXRegisters;
+
+class AUX {
```

Comment:
I'd highly suggest to not use a class here. Instead use a namespace and declare `Memory::TypedMapping<AUXRegisters volatile> m_registers;` in the cpp file so it's "private" to the implementation. Every other method can reside in the namespace which would make it less awkward with `the()::` call dereference each time.

Reply:
I just followed what other driver code in RPi/ does. But I could change it if you think this looks better without the class.

The nice thing about having a singleton class is that the MM region is automatically allocated on first use. But I guess I could also make the `TypedMapping` itself a singleton (that would be a bit weird tho)?

Reply:
A singleton of `TypedMapping` should be OK in my opinion. When you have more interesting things to add, make a structure and include it so that would be the singleton, in the cpp file (and not exported via a header).
The whole `SOMETHING_SINGLETON::the()` concept has proved to be a very clunky way to use global variables and made it so we generated (myself included, and probably most of it) lots of ugly code over the years.

Reply:
I use a private struct for the singleton now, since constructing a `TypedMapping` requires the address as an argument. I tried using a custom InitFunction template parameter, but that won't work, since it expects a raw pointer.

We may actually need to revert to having a class for it once this code uses the devicetree, since the MiniUART code would then probably need to get the AUX via the devicetree node as a key. Or we require that the AUX driver is initialized first somehow.